ferrocene/ci/scripts/setup-venv.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT OR Apache-2.0
+# SPDX-FileCopyrightText: The Ferrocene Developers
+
+set -xeuo pipefail
+
+if command -v uv &> /dev/null; then
+    echo "uv was already installed, skipping"
+    exit 0
+fi
```

Comment:
Is there a case where `uv` could be installed, but no venv has been set up? Iiuc the script currently assumes either both or none are true.

Reply:
I think it's possible that this could happen, yeah. I wasn't really sure how thorough to be since theoretically if this script has run already, it completed wholly (if it failed, the CI job would fail and be rejected). This likely isn't going to be run by us developers.

Reply:
Oki ☺️